scrapper.py

```diff
@@ -31,67 +42,260 @@ class Crawler:
+        self.full_url = full__url
+        self.article_id = article_id
+        self.article = Article(self.full_url, self.article_id)
 
     def _fill_article_with_text(self, article_soup):
-        pass
+        try:
+            text = article_soup.find('div', {'class': 'text letter', 'itemprop': 'articleBody'}).text.strip()
+            text = [i for i in text.split('\n') if 'Фото:' not in i and 'Автор:' not in i
+                    and '© фото:' not in i and 'Источник:' not in i]
+            self.article.text = '\n'.join(text).strip()
+        except AttributeError:
+            print('    unable to parse', self.full_url)
 
     def _fill_article_with_meta_information(self, article_soup):
-        pass
+        try:
+            title = article_soup.title.text
+            self.article.title = title
+
+            credit = article_soup.find('div', {'class': 'credits t-caption'}).text.strip().split('\n')[0]
+            if 'Автор:' in credit:
+                author = article_soup.find('div', {'class': 'credits t-caption'}).text.strip().split('\n')[0][7:]
+            elif 'Источник:' in credit:
+                author = article_soup.find('div', {'class': 'credits t-caption'}).text.strip()
+                author = author.split('\n')[0][9:].strip()
+            else:
+                author = 'Not found'
+            self.article.author = author
+            when = article_soup.find('div', {'class': 'b-caption'}).text.strip().split('\n')[1]
+            self.article.date = self.unify_date_format(when)
+
+            topic = article_soup.find('div', {'class': 'b-caption'}).text.strip().split('\n')[0]
+            self.article.topics = topic
+        except AttributeError:
+            print('    something is off with', self.full_url)
+        # print(title)
+        # self.article.title = title
 
     @staticmethod
     def unify_date_format(date_str):
         """
         Unifies date format
         """
-        pass
+        day, month, year = date_str.split()
+        if len(day) == 1:
+            day = '0' + day
+        match = {'янв': '01',
+                 'фев': '02',
+                 'мар': '03',
+                 'апр': '04',
+                 'май': '05',
+                 'июн': '06',
+                 'июл': '07',
+                 'авг': '08',
+                 'сен': '09',
+                 'окт': '10',
+                 'ноя': '11',
+                 'дек': '12'}
+        return date.fromisoformat(year + '-' + match[month] + '-' + day)
 
     def parse(self):
         """
         Parses each article
         """
-        pass
+        self.article.url = self.full_url
+        self.article.article_id = self.article_id
+        html = requests.get(self.full_url, 'html.parser').text
+        article_bs = BeautifulSoup(html, 'html.parser')
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        self.article.save_raw()
```

Comment:
remember S in SOLID? is it responsibility of **parse** method to **save** the file?